### ISSUE TITLE
support aoe (as storage backend)

### ISSUE THREAD

**Issue by chrstphrchvz** [rockstor/rockstor-core#867]
~~Similar to #636, in being able to use SAN as a storage provider~~ (I'm considering reusing PCs as enclosures rather than acquiring USB enclosures or other direct-attached equipment).

~~Not sure if this is any easier than supporting iSCSI, though:~~ might need to provide aoetools (can't find an official CentOS 7 package, although the package for CentOS 6 seemed to work), and then be able to `aoe-discover` before mounting. Fortunately, the use of Elrepo's kernel-ml provides the aoe driver, which would need to be added to `/etc/modules`.

~~One issue I notice is that `lsblk` (used by btrfs.py) does not list devices in `/dev/etherd/`. I can ask the folks at util-linux if those should be listed; if that's something that can be added to `lsblk`, then aoe targets might show up right away in Rockstor disks.~~

~~At the same time, there's probably not nearly as much demand for aoe support as there is for iSCSI.~~

EDIT: now that I've read what others had posted on the forum about iSCSI, this is different in that I'd like to be able to have Rockstor provide its existing NAS functionality, but using external SAN storage, not to have Rockstor act as a SAN itself.

EDIT: I'll have to try again to see if there's a problem with `lsblk` in the first place. I had tried seeing if it would show devices in `/dev/etherd` from the target; not even `fdisk -l` shows them because what I tested was actually (1) a symbolic link to the file being exported, and (2) smaller than 1GB, which explains why it should not show up in Rockstor's disks.

**Comment by chrstphrchvz** [comment]
There might be an issue in Rockstor that would prevent using aoe targets at all, even without managing them from Rockstor (similar to any other block devices). After running `aoe-discover`, and now having a device '/dev/etherd/e1.2', "Rescan" disks no longer works.

from rockstor.log:
```
[07/Sep/2015 11:13:36] ERROR [storageadmin.util:38] request path: /api/disks/scan method: POST data: <QueryDict: {}>
[07/Sep/2015 11:13:36] ERROR [storageadmin.util:39] exception: value too long for type character varying(10)
Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/rest_framework_custom/generic_view.py", line 40, in _handle_exception
    yield
  File "/opt/rockstor/src/rockstor/storageadmin/views/disk.py", line 107, in post
    return self._scan()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/transaction.py", line 339, in inner
    return func(*args, **kwargs)
  File "/opt/rockstor/src/rockstor/storageadmin/views/disk.py", line 77, in _scan
    dob.save()
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/base.py", line 545, in save
    force_update=force_update, update_fields=update_fields)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/base.py", line 573, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/base.py", line 654, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/base.py", line 687, in _do_insert
    using=using, raw=raw)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/manager.py", line 232, in _insert
    return insert_query(self.model, objs, fields, **kwargs)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/query.py", line 1511, in insert_query
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/models/sql/compiler.py", line 899, in execute_sql
    cursor.execute(sql, params)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/utils.py", line 99, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/opt/rockstor/eggs/Django-1.6.2-py2.7.egg/django/db/backends/util.py", line 53, in execute
    return self.cursor.execute(sql, params)
DataError: value too long for type character varying(10)
```

I'm inclined to think this is an error in handling the output of `lsblk`, which now includes a device with a name greater than 10 characters:
```
[root@rockstor ~]# lsblk -P -o NAME,MODEL,SERIAL,SIZE,TRAN,VENDOR,HCTL,TYPE,FSTYPE,LABEL,UUID
⋮
NAME="etherd!e1.2" MODEL="" SERIAL="" SIZE="1.5G" TRAN="" VENDOR="" HCTL="" TYPE="disk" FSTYPE="btrfs" LABEL="aoetest" UUID="b57ac58b-827f-441e-9323-d4bcc6ef60d9"
```
This error might be of more general concern if there's other storage schemes using devices with long names, although I'm not aware of any (the longest I can think of would be `mmcblk0`).

**Comment by schakrava** [comment]
Thanks for filing this issue @chrstphrchvz . I too share the general concern for longer disk names. Perhaps we could just bump it up to something generous like 64 characters?

**Comment by schakrava** [comment]
The longer disk name issue is now fixed by the commit reference above and merged into 3.8-8.
